src/H5VLnative_dataset.c

```diff
@@ -346,6 +346,31 @@ H5VL__native_dataset_specific(void *obj, H5VL_dataset_specific_t specific_type,
             break;
         }
 
+        case H5VL_DATASET_WAIT: { /* H5Dwait */
+            /* The native VOL connector doesn't support asynchronous
+             *      operations, so this is a no-op.
+             */
+            break;
+        }
```

Comment:
The dataset wait case doesn't belong in 1.12

Reply:
Resolved in [0b97a78](https://github.com/HDFGroup/hdf5/pull/1970/commits/0b97a78647db222e91d0b9a466f5abf8392fb100)

We are now using `H5VL_NATIVE_DATASET_CHUNK_ITER` instead.